diff -uNr Smarty-3.1.5.old/libs/Smarty.class.php Smarty-3.1.5/libs/Smarty.class.php
--- Smarty-3.1.5.old/libs/Smarty.class.php	2011-11-14 18:55:20.000000000 +0100
+++ Smarty-3.1.5/libs/Smarty.class.php	2011-11-15 22:52:22.824193889 +0100
@@ -43,7 +43,7 @@
 * Sets SMARTY_DIR only if user application has not already defined it.
 */
 if (!defined('SMARTY_DIR')) {
-    define('SMARTY_DIR', dirname(__FILE__) . DS);
+    define('SMARTY_DIR', '/usr/share/php/Smarty3/');
 }
 
 /**
